packages/fxa-event-broker/lib/notificationProcessor.ts

```diff
@@ -130,85 +87,92 @@ class ServiceNotificationProcessor {
+    const message = ServiceNotification.from(this.logger, body);
+    if (!message) {
+      // Anything that isn't a message we want
+      this.logger.debug('unwantedMessage', { message });
       return;
     }
     switch (message.event) {
       case LOGIN_EVENT: {
-        let loginMessage: loginSchema;
-        try {
-          loginMessage = joi.attempt(message, LOGIN_SCHEMA);
-        } catch (err) {
-          this.logger.error('badLoginMessage', { err });
-          return;
-        }
-        // Sync and some logins don't emit a clientId, so we have nothing to track
-        if (!loginMessage.clientId) {
-          this.logger.debug('unwantedMessage', { message });
-          return;
-        }
-        await this.db.storeLogin(loginMessage.uid, loginMessage.clientId);
-        this.logger.debug('sqs.loginEvent', loginMessage);
-        return;
+        await this.handleLoginEvent(message);
+        break;
       }
       case SUBSCRIPTION_UPDATE_EVENT: {
-        let subMessage: subscriptionUpdateSchema;
-        try {
-          subMessage = joi.attempt(message, SUBSCRIPTION_UPDATE_SCHEMA);
-        } catch (err) {
-          this.logger.error('badSubscriptionUpdateMessage', { err });
-          return;
-        }
-        const clientIds = await this.db.fetchClientIds(subMessage.uid);
-
-        this.logger.debug('sqs.subEvent', subMessage);
-
-        // Split the product capabilities by clientId each capability goes to
-        const notifyClientIds: { [clientId: string]: string[] } = {};
-        subMessage.productCapabilities
-          .map(item => item.split(':'))
-          .forEach(([clientId, capability]) => {
-            if (notifyClientIds[clientId]) {
-              notifyClientIds[clientId].push(capability);
-            } else {
-              notifyClientIds[clientId] = [capability];
-            }
-          });
-
-        const baseMessage = {
-          capabilities: [],
-          event: subMessage.event,
-          isActive: subMessage.isActive,
-          uid: subMessage.uid
-        };
-
-        Object.entries(notifyClientIds)
-          .filter(([clientId]) => clientIds.includes(clientId))
-          .forEach(async ([clientId, capabilities]) => {
-            const topicName = 'rpQueue-' + clientId;
-            const rpMessage = Object.assign(
-              {},
-              {
-                ...baseMessage,
-                capabilities
-              }
-            );
-
-            // TODO: Failures to publish due to missing queue should be Sentry reported.
-            const messageId = await this.pubsub.topic(topicName).publishJSON(rpMessage);
-            this.logger.debug('publishedMessage', { topicName, messageId });
-          });
-        return;
+        await this.handleSubscriptionEvent(message);
+        break;
+      }
+      case DELETE_EVENT: {
+        await this.handleDeleteEvent(message);
+        break;
       }
+      default:
+        unhandledEventType(message);
```

Comment:
If I understand correctly, this line means that unhandled messages of known type will throw, instead of slip by silently?

Reply:
That creates an exhaustive check, such that the code will not compile if this is reachable.